cloud/migration-stork.md

```diff
@@ -115,14 +117,105 @@ status:
   schedulerStatus: ""
   storageStatus: ""
 ```
-5. Once you apply the above spec you should be able to check the status of the pairing. On a successful pairing, you should 
+5. (EKS Only) When pairing with an EKS cluster, you also need to pass in your
+   AWS credentials which will be used to generate the IAM token. This can be
+   achieved by performing one of the following steps:
+   1. Create a secret and mount in the stork deployment (Secure)
+       1. On the source cluster, create a secret in kube-system namespace with your aws credentials
+       file:
+       ```
+       $ kubectl create secret generic --from-file=$HOME/.aws/credentials -n  kube-system aws-creds
+       secret/aws-creds created
+       ```
+       2. Mount the secret created above in the stork deployment. Run `kubectl edit deployment -n kube-system stork` and make the following updates.
+           1. Add the following under spec.template.spec: 
+           ```yaml
+           volumes:
+           - name: aws-creds
+             secret:
+                  secretName: aws-creds
+           ```
+           2. Add the following under spec.template.spec.containers:
+           ```yaml
+           volumeMounts:
+           - mountPath: /root/.aws/
+             name: aws-creds
+             readOnly: true
+           ```
+       3. Wait for all the stork pods to be in running state after applying the
+          changes: `kubectl get pods -n kube-system -l name=stork`
+   2. Add environment variable to the client authentication spec (Non-secure)
+
+      If you are pairing to an EKS cluster, the generated clusterpair spec will have a section for
+      performing client authentication using the `aws-iam-authenticator`. You can pass in your AWS
+      credentials through environment variables in this spec.
+      An updated spec would look like the following
+      ```yaml
+      exec:
+        apiVersion: client.authentication.k8s.io/v1alpha1
+        env:
+        - name: "AWS_ACCESS_KEY"
+          value: "<your_access_key>
+        - name: "AWS_SECRET_ACCESS_KEY"
+          value: "<your_secret_key>"
+        args:
+        - token
+        - -i
+        - demo-destination-cluster
+        command: aws-iam-authenticator
+      ```
+6. (GKE Only) When pairing with an GKE cluster, you also need to pass in your
+   Google Cloud credentials which will be used to generate the access tokens. This can be
+   achieved by performing all of the following steps:
+   1. Create a service account key using [the guide from Google Cloud](https://cloud.google.com/iam/docs/creating-managing-service-account-keys)
+   and save it as gcs-key.json. You can also create this using the following command: 
+   ```
+   $ gcloud iam service-accounts keys create gcs-key.json \
```

Comment:
One thing I ran into while continuing to use this and test it is I have to add permissions for this service accound to the *destination cluster* that allow it to create things like namespaces etc. It might be a good idea to add a note or an example of this such as

Where `<cluster-role>` has access to create resources, ive used `cluster-admin` for this but their may be more appropriate roles

```
$ kubectl create clusterrolebinding <svc-accnt> --clusterrole=<cluster-role> --user=<svc-accnt>@iam.gserviceaccount.com
```

Reply:
Meeting this PR for now, will nkae further changes in a separate PR